packages/website/src/App.tsx

```diff
@@ -48,6 +48,8 @@ function CloseSnackBarButton(snackbarProviderRef?: React.RefObject<SnackbarProvi
 
 function App(): React.ReactElement {
   const [user, setUser] = useState<User | undefined>();
+  const [searchText, setSearchText] = useState('');
```

Comment:
I think having this state at such a high level on the website will make the code confusing and cause many unnecessary state changes/re-renders (for example, how do we handle when users search in the search bar while viewing their profile or any other page view that does not include the games list).

 I think a simpler approach here is to:
1. on search (user hits enter while searching in the search bar or clicks search iconbutton), we navigate the user to `/games?searchText=<searchText>`. Heres an [example](https://sourcegraph.com/github.com/turnbasedgames/urturn/-/blob/packages/website/src/gameView/gameList/GameList.tsx?L41) where we already do this and here are the [docs](https://reactrouter.com/en/main/hooks/use-navigate) for it.
2. change the game list component to recognize the searchtext query param. Here are [docs ](https://reactrouter.com/en/main/hooks/use-params) for this , and an [example](https://sourcegraph.com/github.com/turnbasedgames/urturn/-/blob/packages/website/src/gameView/gamePlayer/GameInfo.tsx?L21:22) of how we do it now